File: frontend/src/app/shared/models/session.ts
import { User } from './user';
import { Moment } from 'moment';

/**
 * Текущая сессия пользователя
 */
export interface Session {
  /**
   * Id сессии
   */
  id: string;
  /**
   * JWT токен сессии
   */
  token: string;
  /**
   * объект текущего пользователя
   */
  user: User;
  /**
   * Время создания сессии
   */
  timestamp: Moment;
}

File: backend/ISISNotesBackend.Core/Models/Session.cs
using System;

namespace ISISNotesBackend.Core.Models
{
    public class Session
    {
        public string Id { get; set; }
        public string Token { get; set; }
        public User User { get; set; }
        public DateTime Timestamp { get; set; }

        public Session(string id, string token, User user, DateTime timestamp)
        {
            Id = id;
            Token = token;
            User = user;
            Timestamp = timestamp;
        }
    }
}